### PR TITLE
[bitnami/elasticsearch] Update Ginkgo tests

### PR DESCRIPTION
**Pull request by migruiz4**
### Description of the change

This PR updates the Elasticsearch ginkgo tests aiming to improve their stability.

Currently performs the following actions:
    - Get Statefulset
    - Check spec.Replicas == status.readyReplicas
    - Get Service, Container, and Port
    - Create a Job that performs a `PUT /index` action
    - Scale to 0 replicas (which may take some time for the pod to Terminate)
    - Check status.Replicas == 0
    - Scale to 1 replica (which takes, at least, 100 seconds to become ready due to initialDelaySeconds)
    - Create another job that checks the index exists

We are seeing the Ginko tests fail from time to time, especially at the scale-up and scale-down stage.

- By increasing the timeout to 900 seconds (15 min), we give enough time for the minimum 100 seconds it takes for the data replica to become ready.

- By changing the scale up/down replica checks, from `status.availableReplicas` to `status.replicas` when scaling down and `status.readyReplicas` when scaling up. We ensure there are no 'Terminating' pods after scaling down, and pods are 'Ready' when the scale-up action is performed.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/elasticsearch/templates/master/statefulset.yaml [hunk 1]
```diff
@@ -240,6 +240,7 @@ spec:
           livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.master.livenessProbe "enabled") "context" $) | nindent 12 }}
             tcpSocket:
               port: rest-api
+
```
Comment (juan131) [inline bitnami/elasticsearch/templates/master/statefulset.yaml:243]: Unrequired, right?

File: bitnami/elasticsearch/Chart.yaml [hunk 2]
```diff
@@ -34,4 +34,4 @@ maintainers:
 name: elasticsearch
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/elasticsearch
-version: 19.13.0
+version: 19.13.1
```
Comment (juan131) [inline bitnami/elasticsearch/Chart.yaml:37]: Please revert this change after tests are 👍🏼 , there's no reason to bump chart version

**Review by juan131 — CHANGES_REQUESTED** [review]
changes LGTM!

File: bitnami/elasticsearch/templates/master/statefulset.yaml (hunk 1 above)
Comment (migruiz4) [reply bitnami/elasticsearch/templates/master/statefulset.yaml:243]: Yes, it is just a 'Trigger' to force elasticsearch tests to be executed

**Comment by migruiz4** [comment]
Test results [here](https://github.com/bitnami/charts/actions/runs/6389770774/job/17341739262?pr=19685)

**Review by juan131 — APPROVED** [review]
LGTM
